File: python/calendar.py
import  sys
import datetime

today = datetime.date.today()
today_year = today.year
today_month = today.month

def create_calender(year, month):
    """
　　 Create and display a calendar for the specified year and month.
    :param year: The year for the calendar
    :param month: The month for the calendar
    :return:
    """

    print(f"{month}月　{year}")
    print("月  火  水  木  金  土  日")
    first_day = datetime.date(year, month, 1)
    if month == 12:
        last_day = datetime.date(year + 1, 1, 1) - datetime.timedelta(days=1)
    else:
        last_day = datetime.date(year, month + 1, 1) - datetime.timedelta(days=1)

    week_start = first_day - datetime.timedelta(days=first_day.weekday())
    week_end = last_day + datetime.timedelta(days=6-last_day.weekday())

    current_day = week_start

    while current_day <= week_end:
        for _ in range(7):
            if current_day.month == month:
                print(f"{current_day.day:2d}", end="　")
            else:
                print("  ", end="　")
            current_day += datetime.timedelta(days=1)
        print()



def main():
    if len(sys.argv) == 2 and sys.argv[1] == "-m":
        create_calender(today_year, today_month)
    elif len(sys.argv) == 3 and sys.argv[1] == "-m":
        try:
            current_month = int(sys.argv[2])
            if 1 <= current_month <= 12:
                create_calender(today_year, current_month)
            else:
                print(f"{current_month} is neither a month number (1..12) nor a name")
        except ValueError:
            print(f"{current_month} is neither a month number (1..12) nor a name")

if __name__ == "__main__":
    main()

File: javascript/calendar.js
console.log(process.argv);

const today = new Date;

if (process.argv.length === 2){
    year = today.getFullYear();
    month = today.getMonth() + 1;
} else if (process.argv.length === 4 && process.argv[2] === "-m") {
    year = today.getFullYear();
    const inputMonth = process.argv[3];
    const monthNum = parseInt(inputMonth);
    if (isNaN(monthNum)){
        console.log("数字で入力してください");
        process.exit(1);
    } else if (monthNum < 1 || monthNum > 12){
        console.log("1~12の数字を入力してください");
        process.exit(1);
    } 
    month = monthNum;
}

const header = `日 月 火 水 木 金 土`;

const dateInMonth = new Date(year, month, 0).getDate();

const firstDay = new Date(year, month -1, 1);
const startDay = firstDay.getDay();

 let calendarText = " ";
 for (let i = 0; i < 42; i++) {
    if(i < startDay){
        calendarText += " ".padStart(2, " ");
    }else if(i < startDay + dateInMonth){
        const currentDay = i - startDay + 1;
        calendarText += String(currentDay).padStart(2, " ");
    }else{
        calendarText += " ".padStart(2, " ");
    }

    if((i + 1) % 7 === 0){
        calendarText += "\n";
    } else{
        calendarText += " ";
    }
 }

 const calendarDisplay = `    ${year}年${month}月
${header}
${calendarText}`;

console.log(calendarDisplay);